src/java/org/apache/cassandra/cql3/QueryProcessor.java

```diff
@@ -643,8 +655,10 @@ public ResultMessage.Prepared prepare(String queryString, ClientState clientStat
         boolean useNewPreparedStatementBehaviour = useNewPreparedStatementBehaviour();
         MD5Digest hashWithoutKeyspace = computeId(queryString, null);
         MD5Digest hashWithKeyspace = computeId(queryString, clientState.getRawKeyspace());
-        Prepared cachedWithoutKeyspace = preparedStatements.getIfPresent(hashWithoutKeyspace);
-        Prepared cachedWithKeyspace = preparedStatements.getIfPresent(hashWithKeyspace);
+        Pair<Prepared, Integer> cacheValue = preparedStatements.getIfPresent(hashWithoutKeyspace);
+        Prepared cachedWithoutKeyspace = cacheValue == null ? null : cacheValue.left();
+        cacheValue = preparedStatements.getIfPresent(hashWithKeyspace);
+        Prepared cachedWithKeyspace = cacheValue == null ? null : cacheValue.left();
```

Comment:
The introduction of `extractPreparedFromPair ` would allow us to simplify it to:
```
Prepared cachedWithoutKeyspace = extractPreparedFromPair(preparedStatements.getIfPresent(hashWithoutKeyspace));
Prepared cachedWithKeyspace = extractPreparedFromPair(preparedStatements.getIfPresent(hashWithKeyspace));
 ```